build/cmake/LoadTests.cmake

```
@@ -0,0 +1,48 @@
+# This file is include()'d by CTest. It executes test-libmongoc to get a list
+# of all tests that are registered. Each test is then defined as a CTest test,
+# allowing CTest to control the execution, parallelization, and collection of
+# test results.
+
+if (NOT EXISTS "${TEST_LIBMONGOC_EXE}")
+    # This will fail if 'test-libmongoc' is not compiled yet.
+    message (WARNING "The test executable ${TEST_LIBMONGOC_EXE} is not present. "
+                     "Its tests will not be registered")
+    add_test (mongoc/not-found NOT_FOUND)
+    return ()
+endif ()
+
+# Get the list of tests
+execute_process (
+    COMMAND "${TEST_LIBMONGOC_EXE}" --list-tests --no-fork
+    OUTPUT_VARIABLE tests_out
+    WORKING_DIRECTORY "${SRC_ROOT}"
+    RESULT_VARIABLE retc
+    )
+if (retc)
+    # Failed to list the tests. That's bad.
+    message (FATAL_ERROR "Failed to run test-libmongoc to discover tests [${retc}]:\n${tests_out}")
+endif ()
+
+# Split lines on newlines
+string (REPLACE "\n" ";" lines "${tests_out}")
+
+# Generate the test definitions
+foreach (line IN LISTS lines)
+    if (NOT line MATCHES "^/")
+        # Only generate if the line begins with `/`, which all tests should.
+        continue ()
+    endif ()
+    # The new test name is prefixed with 'mongoc'
+    set (test "mongoc${line}")
```

```suggestion
    set (test "${line}")
```
Note: I still do not see the need to break symmetry with `test-libmongoc` by adding a prefix to test names, especially given that distinct test names with the configuration included are proven no longer necessary. I would prefer the `"mongoc"` prefix be removed as well.

---

At least in my VSCode setup, the leading `/` adds an extraneous blank node at the root of the test suite tree. This isn't a blocking issue, but it is confusing to look at, and is mostly aesthetic.